docs/design/log-api.md

```diff
@@ -0,0 +1,148 @@
+# Logs Bridge API
+
+Author: Robert Pająk
+
+Tracking issue at [#4696](https://github.com/open-telemetry/opentelemetry-go/issues/4696).
+
+## Abstract
+
+<!-- A short summary of the proposal. -->
+
+We propose adding a `go.opentelemetry.io/otel/log` Go module which will provide
+[Logs Bridge API](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/).
+
+## Background
+
+<!-- An introduction of the necessary background and the problem being solved by the proposed change. -->
+
+They key challenge is to create a well-performant API compliant with the specification.
+Performance is seen as one of the most imporatant charactristics of logging libraries in Go.
+
+## Design
+
+This proposed design aims to:
+
+- be specification compliant,
+- have similar API to Trace and Metrics API,
+- take advantage of both OpenTelemetry and `slog` experience to achieve acceptable performance.
+
+### LoggerProvider
+
+The [`LoggerProvider` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#loggerprovider)
+is defined as an interface.
+
+```go
+type LoggerProvider interface{
+    Logger(name string, options ...LoggerOption) Logger
+}
+```
+
+### Logger
+
+The [`Logger` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#logger)
+is defined as an interface.
+
+```go
+type Logger interface{
+    Emit(ctx context.Context, options ...RecordOption)
```

Comment:
I think this needs to emit a log record, right?

Reply:
It is the only design choice I am struggling with 😄 

Initially (in my initial version today) it was like you propose `Emit(ctx context.Context, record Record)`. 

This would give the advantage that the SDK would not need to call `NewRecord(options...)`. And the user can still easily create a helper:

```go
func log(ctx context.Context, l Logger, options ...RecordOption) {
    r := log.NewRecord(options...)
    l.Emit(ctx, r)
}
```

The only reason why I not decided to do so is that I felt unsure if anyone would like to reuse a record and I thought that  just passing options would be a more-user friendly API.

I decided with the current proposal because similarly in https://pkg.go.dev/go.opentelemetry.io/otel/metric#Meter . we could have:

```
Int64Counter(name string, cfg Float64HistogramConfig) (Int64Counter, error)
```

instead of 

```go
Int64Counter(name string, options ...Int64CounterOption) (Int64Counter, error)
```

Maybe we should benchmark if the design would affect the performance? I was also a little afraid that declaration with `Emit(ctx context.Context, record Record)` may require one additional copy of `Record`.

Reply:
(I assumed it was just a typo).  Please add it as an alternative considered if this is a design choice :).

Reply:
I will include it in "Rationale".

Reply:
Done